butane_core/src/migrations/fsmigrations.rs

```diff
@@ -271,6 +271,24 @@ impl FsMigrations {
         f.write_all(serde_json::to_string(state)?.as_bytes())
             .map_err(|e| e.into())
     }
+    /// Detach the latest migration from the list of migrations,
+    /// leaving the migration on the filesystem.
+    pub fn detach_latest_migration(&mut self) -> Result<()> {
```

Comment:
Making this method solely an impl on `FsMigrations` rather than a trait method of `MigrationsMut` is inconsistent with how all the other operations methods from the CLI are handled. I think I'd prefer to stick with the existing pattern and make it a trait method for consistency/cleanliness. Since there are no other impls of the trait beyond the in-memory backend used by some tests though, I don't feel super strongly.

Reply:
It was put here as `detach_latest_migration` needs to use `get_state` which only exists on `FsMigrations` and doesnt make sense in the context of `MemMigrations`

Reply:
It might make more sense to rename `detach_latest_migration` to be `rollback_state`, or even `downgrade` if we want similar terminology between removing a migration from the database and removing a migration from the series on the filesystem.

Reply:
Ah that's fair. We can revisit in the future if we end up with other "real" migration backends for some reason.